addons/portal/controllers/mail.py

```diff
@@ -91,12 +103,16 @@ def _message_post_helper(res_model, res_id, message, token='', nosubscribe=True,
 class PortalChatter(http.Controller):
 
     @http.route(['/mail/chatter_post'], type='http', methods=['POST'], auth='public', website=True)
-    def portal_chatter_post(self, res_model, res_id, message, **kw):
+    def portal_chatter_post(self, res_model, res_id, message, attachment_ids=None, **kw):
         url = request.httprequest.referrer
-        if message:
+        if attachment_ids:
```

Comment:
Giving some correct access token the attachment_ids will be read as sudo in message_post, we need to be sure that the attachments are accessible before this step. Unfortunately it is difficult to know. If we ignore the comment below about attachments model and res_id at create, we may want check that attachment res_ids and res_model are matching the record, but it could come from a private note, so we may want to check that the attachment is not linked to any message yet. 
If we link the attachment to mail.compose.message, we could check that model of all attachment_ids is mail.compose.message and res_id is 0. 

So a big security issue here I think, I need to check this in details.

Reply:
@Xavier-Do , we can not use the model 'mail.compose.message' and res_id as 0 because when attachments are added by the internal user. it adds the record's model and res_id on attachment and we want to allow it to download by the public user. (For Ex: while sending a quotation, we attach the quotation pdf and should be accessible by the public user when he access the quotation ) 

as per your first point, we are using the res_id and model and only using those message's attachment which is linked with `website_message_ids` and also checking for the model and res_id at download.  Thanks